### ISSUE TITLE
Dropdown: The selected option is not unselected from the list on clear action

### ISSUE THREAD

**Issue by mikejav** [primefaces/primereact#7576]
### Describe the bug

For the `focusOnHover={false}` and `showClear={true}` props combination when you click "clear" action, the value is reset, but the selected option remain selected on the list.

![gif3](https://github.com/user-attachments/assets/7cc83d2d-c411-44a1-a7a1-581710a7d1f5)

### Reproducer

https://stackblitz.com/edit/fdkzcxdy?file=src%2FApp.jsx

### System Information

```Shell
System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz 
    Memory: 17.13 GB / 31.93 GB
  Binaries:
    Node: 20.10.0 - C:\Program Files\NodeJS 20.10.0\node.EXE
    npm: 10.2.3 - C:\Program Files\NodeJS 20.10.0\npm.CMD
  Browsers:
    Edge: Chromium (131.0.2903.51)
    Internet Explorer: 11.0.19041.4355
  npmPackages:
    primereact: ^10.8.5 => 10.8.5
    react: ^18.2.0 => 18.2.0
```

### Steps to reproduce the behavior

1. Go to https://stackblitz.com/edit/fdkzcxdy?file=src%2FApp.jsx
2. Select any option in the Dropdown
3. Click the "X" (the clear action)
4. Open the Dropdown
6. The previously selected option remains selected on the options list

### Expected behavior

The selected option should be unselected after you click the clear action
